cpp/src/merge/merge.cu

```diff
@@ -493,6 +638,13 @@ table_ptr_type merge(std::vector<table_view> const& tables_to_merge,
 
   CUDF_EXPECTS(key_cols.size() == column_order.size(),
                "Mismatched size between key_cols and column_order");
+  CUDF_EXPECTS(std::accumulate(tables_to_merge.cbegin(),
+                               tables_to_merge.cend(),
+                               cudf::size_type{0},
+                               [](auto const& running_sum, auto const& tbl) {
+                                 return running_sum + tbl.num_rows();
+                               }) <= std::numeric_limits<cudf::size_type>::max(),
+               "Total number of merged rows exceeds row limit");
```

Comment:
@divyegala (cc: @mythrocks) I don't think this actually protects against overflow. By the definition of overflow, the sum will always be less than the max limit of `size_type`. You need to use a _larger_ type and accumulate in `std::size_t` precision. Can you file a follow-up PR with this change?

```suggestion
  CUDF_EXPECTS(std::accumulate(tables_to_merge.cbegin(),
                               tables_to_merge.cend(),
                               size_t{0},
                               [](auto const& running_sum, auto const& tbl) {
                                 return running_sum + static_cast<size_t>(tbl.num_rows());
                               }) <= static_cast<size_t>(std::numeric_limits<cudf::size_type>::max()),
               "Total number of merged rows exceeds row limit");
```

Examples of this pattern:
- https://github.com/rapidsai/cudf/blob/2a923dfff84a8ce3422a126e9d0035282a0bea2f/cpp/src/strings/copying/concatenate.cu#L221
- https://github.com/rapidsai/cudf/blob/2a923dfff84a8ce3422a126e9d0035282a0bea2f/cpp/src/copying/scatter.cu#L324
- https://github.com/rapidsai/cudf/blob/2a923dfff84a8ce3422a126e9d0035282a0bea2f/cpp/src/scalar/scalar.cpp#L70

Reply:
Thanks for pointing that out @bdice! I'll raise a PR